packages/zoe/src/contractFacet/contractFacet.js

```diff
@@ -191,6 +214,69 @@ export function buildRootObject(powers, _params, testJigSetter = undefined) {
       return { zcfSeat, userSeat: userSeatPromiseKit.promise };
     };
 
+    /** @type {MakeInvitation} */
+    const makeInvitation = (
+      offerHandler = () => {},
+      description,
+      customProperties = {},
+    ) => {
+      assert.typeof(
+        description,
+        'string',
+        details`invitations must have a description string: ${description}`,
+      );
+
+      const invitationHandle = makeHandle('InvitationHandle');
+      invitationHandleToHandler.init(invitationHandle, offerHandler);
+      /** @type {Promise<Payment>} */
+      const invitation = E(zoeInstanceAdmin).makeInvitation(
+        invitationHandle,
+        description,
+        customProperties,
+      );
+      return invitation;
+    };
+
+    /** @type {MakeInvitationWithSeat} */
+    const makeInvitationWithSeat = async (
+      offerHandler,
+      description,
+      customProperties,
+      exit,
+    ) => {
+      const invitation = makeInvitation(
+        offerHandler,
+        description,
+        customProperties,
+      );
+      const invitationAmount = await invitationIssuer.getAmountOf(invitation);
+      // AWAIT ///
+
+      const {
+        updater,
+        zoeSeatAdminPromiseKit,
+        userSeatPromiseKit,
+        seatHandle,
+        zcfSeat,
+      } = initializeEmptySeatData(exit);
+      const invitationHandle = invitationAmount.value[0].handle;
+      invitationHandleToZcfSeatKit.init(invitationHandle, {
+        zcfSeatAdmin: zcfSeatToZCFSeatAdmin.get(zcfSeat),
+        zcfSeat,
+      });
+
+      await E(zoeInstanceAdmin)
+        .makeLaterEscrowSeat(seatHandle, invitationHandle)
+        .then(({ zoeSeatAdmin, notifier: zoeNotifier, userSeat }) => {
+          updateFromNotifier(updater, zoeNotifier);
```

Comment:
Oops. When I deprecated `updateFromIterable` in favor of `observeIteration` I should have similarly deprecated `updateFromNotifier` in favor of something like `observeNotifications`. No change needed in this PR since repairing my oversight and fixing callers should be done in a subsequent PR.

Reply:
Recorded as https://github.com/Agoric/agoric-sdk/issues/2049